### PR TITLE
Generate repo pages

### PR DESCRIPTION
**Pull request by glass-ships**
also add logging and utils file for helper methods

### CODE REVIEW

**Review by amc-corey-cox — APPROVED** [review]
This looks great. I can't wait to see what the generated pages look like!

**Comment by matentzn** [comment]
@glass-ships - Sorry I didnt realise all the work you were doing here. I tried to reconcile all conflicts with my major changes to main. Can you finalise this PR in light of that?

One thing, can we keep the python dependencies at most at 3.10 for now?

File: .github/workflows/deploy-docs.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 name: Build and Deploy Docs to GitHub Pages
 on:
   push:
-    branches: [ main ]
+    branches: [ main, gen-repo-page ]
```
Comment (matentzn) [inline .github/workflows/deploy-docs.yaml:4]:
```suggestion
    branches: [ main ]
```

File: .github/workflows/main.yaml [hunk 2]
```diff
@@ -12,7 +12,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: ["3.10"]
+        python-version: ["3.11"]
```
Comment (matentzn) [inline .github/workflows/main.yaml:15]: 3.10?

File: project.Makefile [hunk 3]
```diff
@@ -3,21 +3,19 @@
...
 	$(RUN) python scripts/generate_index.py
 
+genrepodocs:
+	$(RUN) python scripts/generate_repo_pages.py
+
 genschemadoc: $(DOCDIR)
 	$(RUN) gen-doc -d $(DOCDIR)/Documentation-Schema $(SOURCE_SCHEMA_PATH)
 
 build-docs: genpython genschemadoc genindex gen-monarch-overview
+	@rm -f $(DOCDIR)/Documentation-Schema/about.md
```
Comment (matentzn) [inline project.Makefile:18]: why this?

File: pyproject.toml [hunk 4]
```diff
@@ -11,7 +11,7 @@ authors = [
 ]
 
 [tool.poetry.dependencies]
-python = "^3.8"
+python = "^3.10"
```
Comment (matentzn) [inline pyproject.toml:14]: I changed that, maybe need to change back.

**Pull request merged**